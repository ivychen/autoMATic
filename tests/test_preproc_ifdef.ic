/* autoMATic Compiler - Spring 2018 - COMS4115 PLT
 by Ivy Chen ic2389, Nguyen Chi Dung ncd2118,
 Nelson Gomez ng2573, Jimmy O'Donnell jo2474 */

#define MY_VAR 777
#define MY_OTHER_VAR MY_VAR

#ifdef MY_VAR
void f() {
	println(MY_VAR);
}
#end

#ifdef MY_OTHER_VAR
#include "lib/stdlib_test.ic"
#end

#ifdef NUMBER
void g() {
	println("hello");
}

void h(int a) {
	println(NUMBER + a);
}
#end

int main() {
	f();  // should println 777
	g();  // should println hello
	h(5); // should println 105 (100 from stdlib_test.ic)
	return 0;
}
