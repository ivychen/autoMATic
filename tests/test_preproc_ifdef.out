777
hello
105
